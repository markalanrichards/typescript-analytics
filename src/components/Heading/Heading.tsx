import * as React from "react"
import styled from "styled-components"

export interface IProps {
  title: string
}

const H1 = styled.h1`
  padding: 1em;
  text-align: center;
  color: cornflowerblue;
  opacity: 1;

  &:hover {
    transition: opacity 0.6s ease-in-out;
    opacity: 0.2;
  }
`

const Heading = (props: IProps) => <H1>{props.title}!</H1>

export default Heading
