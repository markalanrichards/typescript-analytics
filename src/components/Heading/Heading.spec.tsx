import * as React from "react"
import { configure, shallow, ShallowWrapper, HTMLAttributes } from "enzyme"
import Adapter from "enzyme-adapter-react-16"
import "jest-styled-components"
import * as renderer from "react-test-renderer"

configure({
  adapter: new Adapter()
})

import Heading from "./Heading"

let heading: any

beforeAll(() => (heading = renderer.create(<Heading title="TypeScript" />)))

it("should render Heading with title", () => {
  expect(heading.toJSON()).toMatchSnapshot()
})
it("Snapshot is correct", () => {
  expect(heading.toJSON()).toHaveStyleRule("color", "cornflowerblue")
})
