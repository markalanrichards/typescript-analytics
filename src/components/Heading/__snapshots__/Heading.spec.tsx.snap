// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Heading with title 1`] = `
.c0 {
  padding: 1em;
  text-align: center;
  color: cornflowerblue;
  opacity: 1;
}

.c0:hover {
  -webkit-transition: opacity 0.6s ease-in-out;
  transition: opacity 0.6s ease-in-out;
  opacity: 0.2;
}

<h1
  className="c0"
>
  TypeScript
  !
</h1>
`;
